### ISSUE TITLE
In-app Purchase Issue (attempting to upgrade to OsmAnd Pro)

### ISSUE THREAD

**Issue by jayaddison** [osmandapp/OsmAnd#13654]
### Description
Attempting to upgrade the OsmAnd app to an OsmAnd Pro subscription does not function.

### How to reproduce?

- Navigate from the main menu to "Download Maps"
- Click within the downloads-remaining statusbar (that displays "Free version" and the number of map downloads remaining)
- Scroll down in the "Unlimited map downloads" page until the button "Continue with OsmAnd Pro" button prompt is available
- Click the "Continue with OsmAnd Pro" button and select either the "Monthly" or "Annually" subscription options
- Click the "Complete purchase" button 

Expected: purchase takes place via Google Play payments, and the subscription is enabled

Actual: nothing happens at all; the button temporarily changes colour to a slightly darker shade of blue during the click action, but nothing occurs on-screen (no error dialog, no pop-up, no activation of the Google Play app, etc)

![purchase](https://user-images.githubusercontent.com/55152140/151788348-813538e0-030b-4d3f-986e-23c447c3a402.png)

### Your Environment
OsmAnd Version: OsmAnd 4.1.11, 2021-12-15
Android/iOS version: Android 8.0.0
Device model: Moto e5 Plus

**Comment by vshcherb** [comment]
Unfortunately we can't reproduce the issue. Please double check that your device was logged in with Google Play, so OsmAnd has access to proceed with purchase

**Comment by jayaddison** [comment]
Thank you for taking a look @vshcherb - I've tried this again today, and now a Google Play purchase view slides up from the bottom of the screen (as expected) and I am able to proceed with upgrade.

A difference that I noticed is that the purchase options are now displayed in currency `GBP` -- whereas during this initial issue report they were displayed as `EUR` (as shown in the screenshot).  I'm not sure if/why that would be related to a cause, but either way this issue is resolved/no longer occurring on my device.
